### PR TITLE
Check number validity and send notice if wrong

### PR DESCRIPTION
**Pull request by malex18**
I am not a Ruby coder so I have no idea if the style is good or not.
Don't hesitate to tell me if it's shit x)

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -7,6 +7,10 @@ def edit; end
 
   def update
     batch_number = user_params[:batch_number]&.gsub(/[^\d]/, "")
+    if batch_number.to_i > 2**31 - 1 # max value of a 4-bytes integer column in a database
+      redirect_to settings_path, notice: "Please enter a valid batch number"
+      return
```
Comment (Disco-Chef) [inline app/controllers/users_controller.rb:12]: You can omit this `return` here
The `redirect_to` line above will make a new HTTP request into another controller action, meaning that `return` line would not be reached

Comment (Aquaj) [reply app/controllers/users_controller.rb:12]: Careful. `redirect_to` (much like `render`) does not stop the method's execution — it only skips automatic default rendering. The `Batch.find_or_create` and other lines would still be called if the return wasn't here.

Comment (Disco-Chef) [reply app/controllers/users_controller.rb:12]: Uops, my bad! It's been a while

**Comment by pil0u** [comment]
Option 1: avoid 500 for too large batch (what you did)
Option 2: retrieve the latest batch number (somehow) and set this number as the maximum allowed
Option 3: retrieve all real batches associated with their city (somehow), so that the user only have to select from a long list of batches

**Pull request merged**